### PR TITLE
simplify pushevent structure

### PR DESCRIPTION
**Pull request by ramaraochavali**
Removes redundant fields from `PushEvent` and drives every thing from `PushRequest`

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pilot/pkg/xds/ads.go [hunk 1]
```diff
@@ -575,24 +564,24 @@ func (s *DiscoveryServer) DeltaAggregatedResources(stream discovery.AggregatedDi
 // for large configs. The method will hold a lock on con.pushMutex.
 func (s *DiscoveryServer) pushConnection(con *Connection, pushEv *Event) error {
 	// TODO: update the service deps based on NetworkScope
-	if !pushEv.full {
+	if !pushEv.pushRequest.Full {
```
Comment (hzxuzhonghu) [inline pilot/pkg/xds/ads.go:567]: prefer to get the pushRequest first

File: pilot/pkg/xds/ads_common.go [hunk 2]
```diff
@@ -30,11 +30,11 @@ var configKindAffectedProxyTypes = map[resource.GroupVersionKind][]model.NodeTyp
 // towards the proxy.
 func ConfigAffectsProxy(pushEv *Event, proxy *model.Proxy) bool {
 	// Empty changes means "all" to get a backward compatibility.
-	if len(pushEv.configsUpdated) == 0 {
+	if len(pushEv.pushRequest.ConfigsUpdated) == 0 {
```
Comment (hzxuzhonghu) [inline pilot/pkg/xds/ads_common.go:33]: same here

File: pilot/pkg/xds/ads.go (hunk 1 above)
Comment (ramaraochavali) [reply pilot/pkg/xds/ads.go:567]: Do you mean check for nil or some thing else?

Comment (hzxuzhonghu) [reply pilot/pkg/xds/ads.go:567]: Mostly to remove duplicate `pushEv.pushRequest` prefix when accessing some fields

Comment (ramaraochavali) [reply pilot/pkg/xds/ads.go]: ok. makes sense. Changed

**Pull request merged**